orchagent/neighorch.cpp

```diff
@@ -16,12 +19,232 @@ extern FgNhgOrch *gFgNhgOrch;
 
 const int neighorch_pri = 30;
 
-NeighOrch::NeighOrch(DBConnector *db, string tableName, IntfsOrch *intfsOrch) :
-        Orch(db, tableName, neighorch_pri), m_intfsOrch(intfsOrch)
+static bool send_message(struct nl_sock *socket_p, struct nl_msg *msg_p)
+{
+    int err = 0;
+
+    if (!socket_p)
+    {
+        SWSS_LOG_ERROR("Netlink socket null pointer");
+        return false;
+    }
+
+    if ((err = nl_send_auto(socket_p, msg_p)) < 0)
+    {
+        SWSS_LOG_ERROR("Netlink send message failed, error '%s'", nl_geterror(err));
+        return false;
+    }
+
+    nlmsg_free(msg_p);
+    return true;
+}
+
+NeighOrch::NeighOrch(DBConnector *db, string tableName, IntfsOrch *intfsOrch, FdbOrch *fdbOrch, PortsOrch *portsOrch) :
+        Orch(db, tableName, neighorch_pri), m_intfsOrch(intfsOrch), m_fdbOrch(fdbOrch), m_portsOrch(portsOrch)
 {
+    int err = 0;
+
     SWSS_LOG_ENTER();
+
+    m_fdbOrch->attach(this);
+    gPortsOrch->attach(this);
+
+    m_nl_sock = nl_socket_alloc();
+    if (!m_nl_sock)
+    {
+        SWSS_LOG_ERROR("Netlink socket is NOT allocacted");
+    }
+    else if ((err = nl_connect(m_nl_sock, NETLINK_ROUTE)) < 0)
+    {
+        SWSS_LOG_ERROR("Netlink socket connect failed, error '%s'", nl_geterror(err));
+        nl_socket_free(m_nl_sock);
+        m_nl_sock = NULL;
+    }
+}
+
+NeighOrch::~NeighOrch()
+{
+    if (m_fdbOrch)
+    {
+        m_fdbOrch->detach(this);
+    }
+
+    if (m_nl_sock)
+    {
+        nl_close(m_nl_sock);
+        nl_socket_free(m_nl_sock);
+    }
+}
+
+bool NeighOrch::flushNeighborEntry(const NeighborEntry &entry, const MacAddress &mac)
```

Comment:
@vasant17 , programming kernel from orchagent is not as per the current architecture. I think, this can also cause issue in warmboot. This has been discussed in few other PRs as well. I think we want to have a consistent approach.

Reply:
Addressed!